scripts/cdist

```diff
@@ -20,6 +21,34 @@
 #
 #
 
+def inspect_ssh_mux_opts():
+    import subprocess
+
+    """Inspect whether or not ssh supports multiplexing options"""
+    wanted_mux_opts = {
+        "ControlPath":"~/.ssh/master-%l-%r@%h:%p",
+        "ControlMaster":"auto",
+        "ControlPersist":"125",
+    }
+    # if checked key option is present then this assumes
+    # all options in value are present
+    check = {
+        "ControlMaster": ("ControlMaster", "ControlPath"),
+        "ControlPersist": ("ControlPersist",),
+    }
+    mux_opts = {}
+    for x in check:
+        try:
+            subprocess.check_output("ssh -o {}".format(x),
+                    stderr=subprocess.STDOUT, shell=True)
+        except subprocess.CalledProcessError as e:
+            foo = e.output.decode().lower()
+            if not "bad configuration option" in foo:
+                for o in check[x]:
+                    mux_opts[o] = wanted_mux_opts[o]
+    foo = [" -o {}={}".format(x, mux_opts[x]) for x in mux_opts]
```

Comment:
I think foo might not be a great name and I also think we want to have all or none setup, because just some doesn't really help